web/src/lib/components/photos-page/actions/stack-action.svelte

```diff
@@ -4,56 +4,98 @@
     NotificationType,
     notificationController,
   } from '$lib/components/shared-components/notification/notification';
-  import type { OnStack } from '$lib/utils/actions';
   import { handleError } from '$lib/utils/handle-error';
-  import { updateAssets } from '@immich/sdk';
-  import { getAssetControlContext } from '../asset-select-control-bar.svelte';
-  import { mdiImageMultipleOutline } from '@mdi/js';
+  import { type AssetResponseDto, updateAssets } from '@immich/sdk';
+  import { assetViewingStore } from '$lib/stores/asset-viewing.store';
+  import { getAssetControlContext } from '$lib/components/photos-page/asset-select-control-bar.svelte';
+  import { mdiImageMinusOutline, mdiImageMultipleOutline } from '@mdi/js';
+  import { unstackAssets } from '$lib/utils/asset-utils';
+  import { createEventDispatcher } from 'svelte';
 
-  export let onStack: OnStack | undefined;
+  export let unstack = false;
+
+  const dispatch = createEventDispatcher<{
```

Comment:
As we will update to Svelte 5 in the future, event dispatcher support will be dropped and replaced by call back props, so you can do something like this

```typescript
 export let onClick: ((asset: AssetResponseDto) => void) | undefined = undefined;
```

Reply:
I think you can omit the ` = undefined` at the end